digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriDataImporter.scala

```diff
@@ -108,46 +117,93 @@ class TierekisteriDataImporter(vvhClient: VVHClient, oracleLinearAssetDao: Oracl
         val trLighting = tierekisteriLightingAsset.fetchActiveAssetData(lightingTR, roadNumber)
 
-        trLighting.foreach { tr => println("\nTR: roadNumber, roadPartNumber, start and end: " + tr.roadNumber + " " + tr.roadPartNumber + " " + tr.starMValue + " " + tr.endMValue) }
+        trLighting.foreach { tr => println("\nTR: roadNumber, roadPartNumber, start and end: " + tr.roadNumber + " " + tr.startRoadPartNumber + " " + tr.startAddressMValue + " " + tr.endAddressMValue) }
 
-        val newTRLighting = trLighting.groupBy(trLighting => (trLighting.roadNumber, trLighting.roadPartNumber, trLighting.starMValue, trLighting.endMValue)).map(_._2.head)
+        val newTRLighting = trLighting.groupBy(trLighting => (trLighting.roadNumber, trLighting.startRoadPartNumber, trLighting.startAddressMValue, trLighting.endAddressMValue)).map(_._2.head)
 
         newTRLighting.foreach { trl =>
           OracleDatabase.withDynTransaction {
-            println("\nFetch road addresses to link ids using Viite, trRoadNumber, roadPartNumber start and end " + trl.roadNumber + " " + trl.roadPartNumber + " " + trl.starMValue + " " + trl.endMValue)
-            val roadAddresses = roadAddressDao.getRoadAddressesFiltered(trl.roadNumber, trl.roadPartNumber, trl.starMValue, trl.endMValue)
-            val roadAddressLinks = roadAddresses.map(ra => ra.linkId).toSet
-            val vvhRoadlinks = roadLinkService.fetchVVHRoadlinks(roadAddressLinks).filter(_.administrativeClass == State)
+            println("\nFetch road addresses to link ids using Viite, trRoadNumber, roadPartNumber, start, end and endRoadPartNumber " + trl.roadNumber + " " + trl.startRoadPartNumber + " " + trl.endRoadPartNumber + " " + trl.startAddressMValue + " " + trl.endAddressMValue)
+
+            val roadAddressSections = trl.getRoadAddressSections
+            val roadNumberTRL = roadAddressSections.head.roadNumber
+            val startRoadPartNumberTRL = roadAddressSections.head.roadPartNumber
+            val endRoadPartNumberTRL = roadAddressSections.last.roadPartNumber
+            val startAddressMValueTRL = roadAddressSections.head.startAddressMValue
+            val endAddressMValueTRL_LAST = roadAddressSections.last.endAddressMValue.getOrElse(trl.endAddressMValue)
+            val endAddressMValueTRL_HEAD = roadAddressSections.head.endAddressMValue.getOrElse(trl.endAddressMValue)
+            val trackTRL = roadAddressSections.head.track
+
+            if (roadAddressSections.size == 1) {
+              //Only One Single Part
+              val roadAddressesWithSinglePart = roadAddressDao.getRoadAddress(roadAddressDao.withRoadAddressSinglePart(roadNumberTRL, startRoadPartNumberTRL, trackTRL.value, startAddressMValueTRL, endAddressMValueTRL_HEAD))
+              val roadAddressLinks = roadAddressesWithSinglePart.map(ra => ra.linkId).toSet
+              val vvhRoadlinks = roadLinkService.fetchVVHRoadlinks(roadAddressLinks).filter(_.administrativeClass == State)
```

Comment:
This filter may not do what's wanted. Some VVH road links may have an incorrect administrative class. However, it might be still good if TR may contain municipality lighting info, too.